Arcyd: add file handler to compress rotated debug logs

cs-shadow opened the pull request:
Fixes #18

aevri commented:
This is looking good! I especially like the test driver :)

Please could you also include some detail in the commit message about why you are making the change (could basically copy text from the original issue).

cs-shadow commented:
Used the helper function to write files. Also added a more descriptive commit message. At the time of creating this PR, I was not fully aware of the standards and hence forgot to add the message, should be fine now.

I also added a note that this commit in itself will not change the amount of logs available or stored but will only reduce the size. Increasing the number of rotated logs stored should be a follow up to this.

aevri commented on py/phl/phlsys_compressedlogging__t.py:
```diff
@@ -0,0 +1,123 @@
+
+            phlsys_fs.write_text_file('testfile', content)
+            with phlsys_fs.write_file_lock_context('testfile') as src:
+                src.write(content)
```
I think you accidentally didn't delete lines 76, 77 :)

cs-shadow replied:
Sorry, that was my bad. Should be fine now. :)

aevri commented:
Aha, quick turnaround :) Will look again, thanks!

aevri commented:
Landed at d4c96c5, thanks!
